Blish HUD/Library/Glide/Tween.cs

```diff
@@ -55,348 +55,349 @@ public enum RotationUnit
-			vars = new List<MemberAccessor>();
-			lerpers = new List<MemberLerper>();
-			start = new List<object>();
-			end = new List<object>();
-			behavior = MemberLerper.Behavior.None;
-		}
+            firstUpdate = true;
+            
+            varHash = new Dictionary<string, int>();
+            vars = new List<MemberAccessor>();
+            lerpers = new List<MemberLerper>();
+            start = new List<object>();
+            end = new List<object>();
+            behavior = MemberLerper.Behavior.None;
+        }
 
-		private void AddLerp(MemberLerper lerper, MemberAccessor info, object from, object to)
-		{
-			varHash.Add(info.MemberName, vars.Count);
-			vars.Add(info);
-			
-			start.Add(from);
-			end.Add(to);
-			
-			lerpers.Add(lerper);
-		}
-		
+        private void AddLerp(MemberLerper lerper, MemberAccessor info, object from, object to)
+        {
+            varHash.Add(info.MemberName, vars.Count);
+            vars.Add(info);
+            
+            start.Add(from);
+            end.Add(to);
+            
+            lerpers.Add(lerper);
+        }
+        
         private void Update(float elapsed)
-		{
-        	if (firstUpdate)
-        	{
-        		firstUpdate = false;
-        		
-				var i = vars.Count;
-				while (i --> 0)
-				{
-					if (lerpers[i] != null)
-						lerpers[i].Initialize(start[i], end[i], behavior);
-				}
-        	}
-        	else
-        	{
-				if (this.Paused)
-					return;
-				
-				if (Delay > 0)
-				{
-					Delay -= elapsed;
-					if (Delay > 0)
-						return;
-				}
-				
-				if (time == 0 && timesRepeated == 0 && begin != null)
-					begin();
+        {
+            if (firstUpdate)
+            {
+                firstUpdate = false;
+                
+                var i = vars.Count;
+                while (i --> 0)
+                {
+                    if (lerpers[i] != null)
+                        lerpers[i].Initialize(start[i], end[i], behavior);
+                }
+            }
+            else
+            {
+                if (this.Paused)
+                    return;
+                
+                if (Delay > 0)
+                {
+                    Delay -= elapsed;
+                    if (Delay > 0)
+                        return;
+                }
+                
+                if (time == 0 && timesRepeated == 0 && begin != null)
+                    begin();
+
+                bool doComplete = false;
 
-				time += elapsed;
-				float setTimeTo = time;
-				float t = time / Duration;
-				bool doComplete = false;
-				
-				if (time >= Duration)
-				{
-					if (repeatCount != 0)
-					{
-						setTimeTo = 0;
-						Delay = repeatDelay;
-						timesRepeated++;
-						
-						if (repeatCount > 0) {
+                float t;
+                time += elapsed;
+                if (time >= Duration)
+                {
+                    t = 1;
+                    if (repeatCount != 0)
+                    {
+                        time = 0;
```

Comment:
When reseting time, the right way should be:
```csharp
time -= Duration;
```
By doing it this way, you don't lose time precision.